fia_api/core/utility.py

```diff
@@ -34,3 +38,22 @@ def filter_script_for_tokens(script: str) -> str:
     ]
 
     return "\n".join(filtered_script_list)
+
+
+def safe_check_filepath(filepath: Path, base_path: Path) -> None:
+    """
+    Check to ensure the path does contain the base path and that it does not resolve to some other directory
+    :param filepath: the filepath to check
+    :param base_path: base path to check against
+    :return:
+    """
+    try:
+        filepath.resolve(strict=True)
+        if not filepath.is_relative_to(base_path):
+            raise HTTPException(status_code=HTTPStatus.FORBIDDEN, detail="Invalid path being accessed.")
+    except FileNotFoundError as err:
```

Comment:
Would it be possible to test this error does infact get raised? and that the subsequent execution branches are covered in tests.

Reply:
Can we have a explicit test?